name: Vale linting

on:
  pull_request:
    types: [opened, synchronize, reopened]
    paths:
      - 'website/docs/**/*'
      - 'website/blog/**/*'
      - 'website/**/*'

jobs:
  vale:
    name: Vale linting
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v3
        with:
          fetch-depth: 1

      - name: List repository contents
        run: |
          pwd
          ls -R

      - name: Set up Python
        uses: actions/setup-python@v5
        with:
          python-version: '3.x'

      - name: Install Vale
        run: pip install vale==2.27.0  # Install a stable version of Vale

      - name: Get changed files
        id: changed-files
        uses: tj-actions/changed-files@v34
        with:
          files: |
            website/**/*.md
          separator: ' '

      - name: Run Vale with reviewdog
        if: ${{ steps.changed-files.outputs.any_changed == 'true' }}
        id: vale-lint
        uses: errata-ai/vale-action@reviewdog
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          reporter: github-check
          level: warning  # Only flag warnings
          filter_mode: added  # Run on changed lines only
          files: ${{ steps.changed-files.outputs.all_changed_and_modified_files }}
          separator: ' '
          version: '2.27.0'

      - name: Post warning in the job output if Vale issues found
        if: ${{ steps.vale-lint.outputs.github_check.conclusion != 'success' }}
        run: |
          echo "::warning::Vale linting has detected warnings. Please check the details in the **Files changed** tab or the **Checks** tab."
